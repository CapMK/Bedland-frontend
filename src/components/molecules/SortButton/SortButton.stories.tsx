import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import SortButton from './SortButton';

const meta = {
  title: 'Molecules/SortButton',
  component: SortButton,
  tags: ['autodocs'],
  parameters: {
    backgrounds: { default: 'dark' },
  },
  argTypes: {
    text: {
      description: 'Text displayed in the button',
    },
    changeSortOption: {
      table: {
        disable: true,
      },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof SortButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const SortBuildings: Story = {
  args: {
    text: 'Sort Buildings',
    changeSortOption: () => null,
  },
};
