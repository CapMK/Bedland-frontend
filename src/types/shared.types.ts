export type Getters<T extends string, K> = {
  [k in T]: K;
};

export type Building = {
  id: string;
  name: string;
  address: string;
  managerId: string;
  floors: string;
  createDate: string;
  updateDate: string;
  version: number;
};

export type Flat = {
  id: string;
  version: number;
  createDate: string;
  updateDate: string;
  buildingId: string;
  number: string;
  floor: string;
  shapePath: string;
};

export type FlatWithAddress = Flat & { address: string; city: string };

export type ResidentsFlats = {
  flatId: string;
  flatNumber: string;
  flatAddress: string;
  flatCreateDate: string;
};

export type BuildingWithCity = Building & { city: string };

export type ResidentsFlatsWithCity = ResidentsFlats & { city: string };
