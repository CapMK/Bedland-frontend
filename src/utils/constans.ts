export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
