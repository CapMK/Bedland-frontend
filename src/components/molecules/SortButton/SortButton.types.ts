import { Building } from 'types/shared.types';
import { DisplayListType } from 'components/organisms/DisplayList/DisplayList.types';

export const sortOptions = ['name', 'city', 'date added'] as const;

export type sortType = (typeof sortOptions)[number];

export type SortButtonProps = {
  text: string;
  changeSortOption: (sortOption: sortType) => void;
};

export const IsBuildingGuard = (list: DisplayListType): list is Building[] =>
  (list as Building[])[0].name !== undefined;
