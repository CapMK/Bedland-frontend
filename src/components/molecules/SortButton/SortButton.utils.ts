import { DisplayListType } from 'components/organisms/DisplayList/DisplayList.types';
import getCity from 'components/organisms/DisplayList/DisplayList.utils';
import { IsBuildingGuard, sortOptions, sortType } from './SortButton.types';

export const sortList = (list: DisplayListType, sortOption: sortType): DisplayListType => {
  switch (sortOption) {
    case sortOptions[0]:
      if (IsBuildingGuard(list)) {
        return list.sort((a, b) => a.name.localeCompare(b.name));
      }
      return list.sort((a, b) => a.flatId.localeCompare(b.flatNumber));
    case sortOptions[1]:
      if (IsBuildingGuard(list)) {
        return list.sort((a, b) => getCity(a.address).localeCompare(getCity(b.address)));
      }
      return list.sort((a, b) => getCity(a.flatAddress).localeCompare(getCity(b.flatAddress)));
    case sortOptions[2]:
      if (IsBuildingGuard(list)) {
        return list.sort(
          (a, b) => new Date(a.createDate).getTime() - new Date(b.createDate).getTime(),
        );
      }
      return list.sort(
        (a, b) => new Date(a.flatCreateDate).getTime() - new Date(b.flatCreateDate).getTime(),
      );
    default:
      return list;
  }
};

export const listToSort: DisplayListType = [
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3, 22-233 Poznan',
    managerId: '1',
    floors: '7',
    version: 1,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
  },
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21, 22-233 Wroclaw',
    managerId: '1',
    floors: '5',
    version: 5,
    createDate: '2019-10-03T00:00:00',
    updateDate: '2023-01-01T00:00:00',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51, 22-233 Gdansk',
    managerId: '2',
    floors: '5',
    version: 2,
    createDate: '2019-10-01T00:00:00',
    updateDate: '2020-05-07T00:00:00',
  },
];

export const listSortedByName: DisplayListType = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21, 22-233 Wroclaw',
    managerId: '1',
    floors: '5',
    version: 5,
    createDate: '2019-10-03T00:00:00',
    updateDate: '2023-01-01T00:00:00',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3, 22-233 Poznan',
    managerId: '1',
    floors: '7',
    version: 1,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51, 22-233 Gdansk',
    managerId: '2',
    floors: '5',
    version: 2,
    createDate: '2019-10-01T00:00:00',
    updateDate: '2020-05-07T00:00:00',
  },
];

export const listSortedByCity: DisplayListType = [
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51, 22-233 Gdansk',
    managerId: '2',
    floors: '5',
    version: 2,
    createDate: '2019-10-01T00:00:00',
    updateDate: '2020-05-07T00:00:00',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3, 22-233 Poznan',
    managerId: '1',
    floors: '7',
    version: 1,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
  },
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21, 22-233 Wroclaw',
    managerId: '1',
    floors: '5',
    version: 5,
    createDate: '2019-10-03T00:00:00',
    updateDate: '2023-01-01T00:00:00',
  },
];

export const listSortedByDateAdded: DisplayListType = [
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51, 22-233 Gdansk',
    managerId: '2',
    floors: '5',
    version: 2,
    createDate: '2019-10-01T00:00:00',
    updateDate: '2020-05-07T00:00:00',
  },
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21, 22-233 Wroclaw',
    managerId: '1',
    floors: '5',
    version: 5,
    createDate: '2019-10-03T00:00:00',
    updateDate: '2023-01-01T00:00:00',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3, 22-233 Poznan',
    managerId: '1',
    floors: '7',
    version: 1,
    createDate: '2023-02-14T00:00:00',
    updateDate: '2023-02-14T00:00:00',
  },
];

export const flatsListToSort: DisplayListType = [
  {
    flatId: '2',
    flatNumber: '2',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-12T00:00:00',
  },
  {
    flatId: '1',
    flatNumber: '1',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-14T00:00:00',
  },
  {
    flatId: '5',
    flatNumber: '4',
    flatAddress: 'ul. Kolorowa 6, 60-198 Poznań',
    flatCreateDate: '2023-02-13T00:00:00',
  },
];

export const flatsListSortedByName: DisplayListType = [
  {
    flatId: '1',
    flatNumber: '1',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-14T00:00:00',
  },
  {
    flatId: '2',
    flatNumber: '2',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-12T00:00:00',
  },
  {
    flatId: '5',
    flatNumber: '4',
    flatAddress: 'ul. Kolorowa 6, 60-198 Poznań',
    flatCreateDate: '2023-02-13T00:00:00',
  },
];

export const flatsListSortedByCity: DisplayListType = [
  {
    flatId: '5',
    flatNumber: '4',
    flatAddress: 'ul. Kolorowa 6, 60-198 Poznań',
    flatCreateDate: '2023-02-13T00:00:00',
  },
  {
    flatId: '1',
    flatNumber: '1',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-14T00:00:00',
  },
  {
    flatId: '2',
    flatNumber: '2',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-12T00:00:00',
  },
];

export const flatsListSortedByDateAdded: DisplayListType = [
  {
    flatId: '2',
    flatNumber: '2',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-12T00:00:00',
  },
  {
    flatId: '5',
    flatNumber: '4',
    flatAddress: 'ul. Kolorowa 6, 60-198 Poznań',
    flatCreateDate: '2023-02-13T00:00:00',
  },
  {
    flatId: '1',
    flatNumber: '1',
    flatAddress: 'ul. Legnicka 48H, 54-202 Wrocław',
    flatCreateDate: '2023-02-14T00:00:00',
  },
];
