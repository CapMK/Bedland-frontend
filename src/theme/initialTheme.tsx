import { Theme } from './theme.types';

const initialTheme: Theme = {
  palette: {
    sidebarTabsPressed: '#919AFF',
    blueDark: '#4553DD',
    blueLight: '#6573FF',
    orangeLight: '#FF7C33',
    orangeDark: '#F56513',
    light: '#FCFEFF',
    greyLight: '#F5F6F8',
    greyDark: '#767676',
    greenLight: '#83E57A',
    greenDark: '#30BA57',
    redLight: '#FF4438',
    sidebarTabs: '#CACACA',
    redDark: '#CF0000',
    sidebar: '#0A1122',
    footer: '#3F4476',
    white: '#fff',
    inputGrey: '#4B4B4B',
    widgetsShadows: '#D9D9D9',
    text: '#1B1A1A',
    yellow: '#FFD541',
  },
  typography: {
    header1: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 600,
      fontSize: '2rem',
      letterSpacing: '2px',
      lineHeight: 1.2,
    },
    header2: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 700,
      fontSize: '1.5rem',
      lineHeight: 1.4,
    },
    subtitles: {
      fontFamily: 'Poppins, sans-serif',
      fontWeight: 500,
      fontSize: '1.25rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header3: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 700,
      fontSize: '1.25rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header4: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontSize: '1rem',
      fontWeight: 600,
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    header5: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 600,
      fontSize: '0.875rem',
      lineHeight: 1.4,
      letterSpacing: '0',
    },
    subHeader: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.75rem',
      lineHeight: 1.4,
      letterSpacing: '0',
    },
    paragraph: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: 1.6,
      letterSpacing: '0',
    },
    dataInputsAndTooltips: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.875rem',
      lineHeight: 1.4,
      letterSpacing: '1px',
    },
    unerlined: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 400,
      fontSize: '0.875rem',
      lineHeight: 1.6,
      textDecoration: 'underline',
    },
    captions: {
      fontFamily: 'Source Sans Pro, sans-serif',
      fontWeight: 600,
      fontSize: '0.75rem',
      lineHeight: 1.6,
      letterSpacing: '1px',
    },
  },
};

export default initialTheme;
