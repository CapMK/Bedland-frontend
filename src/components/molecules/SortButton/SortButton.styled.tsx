import styled from 'styled-components';
import { PaletteType } from 'theme/theme.types';

export const DropdownBox = styled.div`
  position: relative;
`;

export const SortDropdown = styled.div<{ palette: PaletteType }>`
  background: ${({ palette }) => palette.white};
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  gap: 0.625rem;
  margin-top: 0.75rem;
  padding: 0.5rem 0.375rem;
  position: absolute;
  width: 100%;
  z-index: 1;
`;

export const SortItem = styled.div<{ palette: PaletteType; active: boolean }>`
  background: ${({ palette, active }) => (active ? palette.blueDark : palette.white)};
  border-radius: 8px;
  padding: 0.563rem 0;
  text-align: center;

  :hover {
    background: ${({ palette, active }) => !active && `${palette.blueDark}15`};
    cursor: pointer;
  }
`;
