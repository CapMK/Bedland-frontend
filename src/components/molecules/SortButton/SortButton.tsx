import { useState, useEffect, useRef, useContext } from 'react';
import Button from 'components/atoms/Button/Button';
import { ThemeContext } from 'theme/ThemeContext';
import Typography from 'components/atoms/Typography/Typography';
import * as Styled from './SortButton.styled';
import { SortButtonProps, sortOptions, sortType } from './SortButton.types';

const SortButton = ({ text, changeSortOption }: SortButtonProps) => {
  const { palette } = useContext(ThemeContext);
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const [activeSortOption, setActiveSortOption] = useState<sortType>('name');
  const ref = useRef<HTMLDivElement | null>(null);

  const handleDropdown = (): void => {
    setIsOpen(!isOpen);
  };

  const onClickOutside = () => {
    setIsOpen(false);
  };

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent): void => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        onClickOutside();
      }
    };

    document.addEventListener('click', handleClickOutside, true);

    return () => {
      document.removeEventListener('click', handleClickOutside, true);
    };
  });

  const handleSortClick = (sortIndex: sortType): void => {
    setActiveSortOption(sortIndex);
    changeSortOption(sortIndex);
    onClickOutside();
  };

  return (
    <Styled.DropdownBox ref={ref}>
      <Button
        variant='primary'
        color='white'
        hover='text'
        icon='arrowDown'
        iconSize={12}
        onClick={handleDropdown}
      >
        {text}
      </Button>

      {isOpen ? (
        <Styled.SortDropdown palette={palette}>
          {sortOptions.map((item) => (
            <Styled.SortItem
              palette={palette}
              active={item === activeSortOption}
              onClick={() => handleSortClick(item)}
              key={item}
            >
              <Typography variant='header5' color={item === activeSortOption ? 'white' : 'text'}>
                {item.charAt(0).toUpperCase() + item.slice(1)}
              </Typography>
            </Styled.SortItem>
          ))}
        </Styled.SortDropdown>
      ) : null}
    </Styled.DropdownBox>
  );
};

export default SortButton;
