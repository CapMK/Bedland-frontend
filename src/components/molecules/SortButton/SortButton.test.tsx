import { renderWithProviders } from 'utils/test.utils';
import { fireEvent, screen, cleanup } from '@testing-library/react';
import initialTheme from 'theme/initialTheme';
import { vi } from 'vitest';
import SortButton from './SortButton';
import {
  flatsListSortedByCity,
  flatsListSortedByDateAdded,
  flatsListSortedByName,
  flatsListToSort,
  listSortedByCity,
  listSortedByDateAdded,
  listSortedByName,
  listToSort,
  sortList,
} from './SortButton.utils';
import { sortOptions } from './SortButton.types';

const mockSort = vi.fn();

describe('SortButton', () => {
  afterEach(cleanup);

  const componentSortButton = () =>
    renderWithProviders(<SortButton text='Sort Buildings' changeSortOption={mockSort} />, {});

  it('renders correctly', () => {
    const snapshot = renderWithProviders(
      <SortButton text='Sort Buildings' changeSortOption={mockSort} />,
      {},
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('dropdown opens and closes when button is clicked', () => {
    componentSortButton();
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    fireEvent.click(sortButton);
    const dropdownOpen = screen.getByText(/date added/i);
    expect(dropdownOpen).toBeTruthy();
    fireEvent.click(sortButton);
    const dropdownClosed = screen.queryByText(/date added/i);
    expect(dropdownClosed).toBeFalsy();
  });

  it('dropdown closes when sorting option is clicked', () => {
    componentSortButton();
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    fireEvent.click(sortButton);
    const sortOption = screen.getByText(/date added/i);
    fireEvent.click(sortOption);
    const dropdownClosed = screen.queryByText(/date added/i);
    expect(dropdownClosed).toBeFalsy();
  });

  it('sorting option is active after selecting it and opening dropdown again', () => {
    componentSortButton();
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    fireEvent.click(sortButton);
    const sortOption = screen.getByText(/date added/i);
    fireEvent.click(sortOption);
    fireEvent.click(sortButton);
    const activeSortOption = screen.getByText(/date added/i).parentElement;
    expect(activeSortOption).toHaveStyle(`background: ${initialTheme.palette.blueDark}`);
  });

  it('dropdown closes when a click occurs anywhere on the site', () => {
    renderWithProviders(
      <>
        <SortButton text='Sort Buildings' changeSortOption={mockSort} />
        <p data-testid='click-anywhere'>click anywhere</p>
      </>,
      {},
    );
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    const clickAnywhere = screen.getByTestId('click-anywhere');
    fireEvent.click(sortButton);
    fireEvent.click(clickAnywhere);
    const dropdownClosed = screen.queryByText(/date added/i);
    expect(dropdownClosed).toBeFalsy();
  });

  it.each([
    [sortOptions[0], listSortedByName],
    [sortOptions[1], listSortedByCity],
    [sortOptions[2], listSortedByDateAdded],
  ])('sortList correctly sorts buildings list by %s', (sortOption, expected) => {
    expect(sortList(listToSort, sortOption)).toEqual(expected);
  });

  it.each([
    [sortOptions[0], flatsListSortedByName],
    [sortOptions[1], flatsListSortedByCity],
    [sortOptions[2], flatsListSortedByDateAdded],
  ])('sortList correctly sorts flats list by %s', (sortOption, expected) => {
    expect(sortList(flatsListToSort, sortOption)).toEqual(expected);
  });
});
