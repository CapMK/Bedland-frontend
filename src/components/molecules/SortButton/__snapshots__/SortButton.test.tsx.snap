// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortButton > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c1:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

.c0 {
  position: relative;
}

<body>
    <div>
      <div
        class="c0"
      >
        <button
          class="c1"
        >
          <svg
            data-testid="arrowDown-icon"
            fill="none"
            height="12"
            viewBox="0 0 12 7"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <mask
              id="mask-arrowDown"
            />
            <path
              d="M10.6667 1.33325L6.08333 5.49992L1.5 1.33325"
              stroke="#1B1A1A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.67"
            />
          </svg>
          Sort Buildings
        </button>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c1:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

.c0 {
  position: relative;
}

<div>
    <div
      class="c0"
    >
      <button
        class="c1"
      >
        <svg
          data-testid="arrowDown-icon"
          fill="none"
          height="12"
          viewBox="0 0 12 7"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="mask-arrowDown"
          />
          <path
            d="M10.6667 1.33325L6.08333 5.49992L1.5 1.33325"
            stroke="#1B1A1A"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.67"
          />
        </svg>
        Sort Buildings
      </button>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
